### ISSUE TITLE
feat(training): FastSACTrainer provider — SAC + distributional critic + autotune alpha (Holosoma port)

### ISSUE THREAD

**Issue by cagataycali** [strands-labs/robots#859]
Part of #858 epic. Parallel track with #C4 (FastTD3). Depends on #C1, #C2, #C3.

## Goal
Port Holosoma's `FastSACAgent` (and AD58-3104's `FastSAC_standalone`
rewrite) into `strands_robots.training.fastsac.FastSACTrainer`:

```python
trainer = create_trainer(\"fastsac\")
spec = TrainSpec(env_name=\"G1JoystickFlatTerrain\",
                 steps=50_000,
                 extra={\"num_envs\": 4096, \"buffer_size\": 1024})
trainer.train(spec)
```

## Algorithm features (from Holosoma `algo.py::fast_sac`)
- SAC with squashed Gaussian actor (`use_tanh=True`, `log_std_min/max`).
- **Distributional critic (C51)**: `num_atoms=101`, `v_min=-20`, `v_max=20`.
- **N Q-networks** (default 2), CDQ-style min reduction.
- **Autotune temperature** (`use_autotune=True`, `target_entropy_ratio`).
- **Symmetry augmentation** (`use_symmetry=True`) — augments
  obs/actions via mirror transform; needs `SymmetryUtils` (#C11 task spec).
- **AMP bf16 + torch.compile**.
- **Empirical obs normalization** (actor + critic separately).
- **Optional CNN encoder** for perception observations
  (`use_cnn_encoder`, `encoder_obs_key=\"perception_obs\"`).
- **Layer norm** in actor + critic (`use_layer_norm=True`).
- **Gradient clipping**, **weight decay**, **fused AdamW with beta=(0.9, 0.95)**.
- **Per-term reward logging** (Holosoma's `reward_log_buffers`).

## Hyperparam presets to port
- G1-29DOF locomotion (Holosoma's headline `exp:g1-29dof-fast-sac`)
- T1 locomotion
- Whole-body tracking (LAFAN / OMOMO) — see #C12

## Acceptance
- [ ] `FastSACTrainer` registered in `registry/policies.json`
- [ ] Matches Holosoma's loss curves on G1 IsaacGym within 5% over 50k iters
- [ ] `trainer.export()` writes ONNX + state_dict (for #C13 deploy)
- [ ] Per-term reward W&B logging on by default

## Refs
- Holosoma FastSAC: `src/holosoma/holosoma/agents/fast_sac/`
- Standalone rewrite (cleaner): `AD58-3104/FastSAC_standalone/isaaclab_fast_sac/`
- FastSAC paper: https://arxiv.org/abs/2512.01996
- FastSAC blog (Arth Shukla): https://arthshukla.substack.com/p/speeding-up-sac-with-massively-parallel
- SAC on massive parallel sim (Antonin Raffin): https://araffin.github.io/post/sac-massive-sim/

**Comment by cagataycali** [comment]
Closing as out of scope for 1.0, and folding the tracking back into the parent epic #858.

The 0.5 -> 1.0 roadmap has six themes - tree-shaking, the dashboard, sealing the policy-provider list, a docs/examples rework, a starter kit and the Sprout embodiment. An off-policy RL training stack with parallel-sim backends (IsaacLab, MJWarp, MuJoCo Playground, HumanoidBench) is not among them, and it is a large enough body of work that keeping fifteen open children makes the 1.0 board unreadable.

Measured before closing, so this is not being written off as already-done: four of these children look shipped if you grade them by whether a class exists, and none of them meets its own acceptance criteria. `FastSacTrainer`, `PpoTrainer`, `VecSimEnv` and `elastic_launch_callable` are all present, but `num_atoms`/`c51`/`v_min` appear nowhere (so #859's distributional critic did not land), `adaptive_kl`/`desired_kl` appear nowhere (#865), `mjwarp` appears nowhere (so #863's vec-env does not span the three backends it names), and `elastic_launch_callable` has zero references from `training/rl/` (#870 asked for a launcher FOR the RL trainers). A symbol is a claim; the criteria are the contract.

#858 stays open as the single post-1.0 tracker for this stack. Re-file any individual piece against the 1.0+ roadmap when the RL lane is scheduled.

Closed as part of the 0.5 -> 1.0 backlog cleanup.
